[Tech Debt]: `TestExpandListOfStringEnum` panics intermittently

### Description


https://github.com/hashicorp/terraform-provider-aws/actions/runs/8627063250/job/23646483202

```
--- FAIL: TestExpandListOfStringEnum (0.00s)
    --- FAIL: TestExpandListOfStringEnum/valid_value (0.00s)
panic: reflect: slice index out of range [recovered]
	panic: reflect: slice index out of range

goroutine 247 [running]:
testing.tRunner.func1.2({0x99c8e0, 0xdd46e0})
	/opt/hostedtoolcache/go/1.21.8/x64/src/testing/testing.go:1545 +0x2[38](https://github.com/hashicorp/terraform-provider-aws/actions/runs/8627063250/job/23646483202?pr=36823#step:7:39)
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.21.8/x64/src/testing/testing.go:1548 +0x[39](https://github.com/hashicorp/terraform-provider-aws/actions/runs/8627063250/job/23646483202?pr=36823#step:7:40)7
panic({0x99c8e0?, 0xdd46e0?})
	/opt/hostedtoolcache/go/1.21.8/x64/src/runtime/panic.go:914 +0x21f
reflect.Value.Index({0x996a00?, 0xc000804180?, 0xc000807520?}, 0xc000521760?)
	/opt/hostedtoolcache/go/1.21.8/x64/src/reflect/value.go:1424 +0x167
github.com/google/go-cmp/cmp.(*state).compareSlice.func1(0x0, 0xffffffffffffffff)
	/home/runner/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:442 +0x93
github.com/google/go-cmp/cmp.(*state).compareSlice(0xc0005325a0, {0xde2c38, 0x996a00}, {0x996a00?, 0xc000804180?, 0x7f14b5eae180?}, {0x996a00?, 0xc000804198?, 0xc000521a28?})
	/home/runner/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:464 +0x558
github.com/google/go-cmp/cmp.(*state).compareAny(0xc0005325a0, {0xdd97b0, 0xc000025000?})
	/home/runner/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:291 +0xbdb
github.com/google/go-cmp/cmp.(*state).comparePtr(0xc0005325a0, {0xde2c38, 0x996a[40](https://github.com/hashicorp/terraform-provider-aws/actions/runs/8627063250/job/23646483202?pr=36823#step:7:41)}, {0x996a40?, 0xc00080[41](https://github.com/hashicorp/terraform-provider-aws/actions/runs/8627063250/job/23646483202?pr=36823#step:7:42)80?, 0x16?}, {0x996a40?, 0xc000804198?, 0x16?})
	/home/runner/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:566 +0x391
github.com/google/go-cmp/cmp.(*state).compareAny(0xc0005325a0, {0xdd9840, 0xc000536480?})
	/home/runner/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:295 +0xb92
github.com/google/go-cmp/cmp.Diff({0x996a40, 0xc000804180}, {0x996a40, 0xc000804198}, {0x0?, 0xc000804180?, 0x0?})
	/home/runner/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:122 +0x75
github.com/hashicorp/terraform-provider-aws/internal/framework/flex.runAutoExpandTestCases.func1(0xc000807380)
	/home/runner/work/terraform-provider-aws/terraform-provider-aws/internal/framework/flex/auto_expand_test.go:909 +0x1b8
testing.tRunner(0xc000807380, 0xc000[46](https://github.com/hashicorp/terraform-provider-aws/actions/runs/8627063250/job/23646483202?pr=36823#step:7:47)c0e0)
	/opt/hostedtoolcache/go/1.21.8/x64/src/testing/testing.go:1595 +0xff
created by testing.(*T).Run in goroutine 11
	/opt/hostedtoolcache/go/1.21.8/x64/src/testing/testing.go:16[48](https://github.com/hashicorp/terraform-provider-aws/actions/runs/8627063250/job/23646483202?pr=36823#step:7:49) +0x3ad
FAIL	github.com/hashicorp/terraform-provider-aws/internal/framework/flex	0.026s
```

### References

- Relates #36789 

### Would you like to implement a fix?

Yes

Race condition is present due to reuse of a pointer to the same slice across test cases.

```
% go test -v -race ./internal/framework/flex/... -run="TestExpandListOfStringEnum"
=== RUN   TestExpandListOfStringEnum
=== PAUSE TestExpandListOfStringEnum
=== CONT  TestExpandListOfStringEnum
=== RUN   TestExpandListOfStringEnum/valid_value
=== PAUSE TestExpandListOfStringEnum/valid_value
=== RUN   TestExpandListOfStringEnum/empty_value
=== PAUSE TestExpandListOfStringEnum/empty_value
=== RUN   TestExpandListOfStringEnum/null_value
=== PAUSE TestExpandListOfStringEnum/null_value
=== CONT  TestExpandListOfStringEnum/valid_value
=== CONT  TestExpandListOfStringEnum/null_value
=== CONT  TestExpandListOfStringEnum/empty_value
==================
WARNING: DATA RACE
Write at 0x00c00051c780 by goroutine 19:
  internal/reflectlite.maplen()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/runtime/map.go:1423 +0x7c
  reflect.Value.Set()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/reflect/value.go:2330 +0x144
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.autoExpander.listOfString()
      /Users/jaredbaker/development/terraform-provider-aws/internal/framework/flex/auto_expand.go:393 +0x74c
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.autoExpander.list()
      /Users/jaredbaker/development/terraform-provider-aws/internal/framework/flex/auto_expand.go:352 +0x414
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.autoExpander.convert()
      /Users/jaredbaker/development/terraform-provider-aws/internal/framework/flex/auto_expand.go:85 +0x6cc
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.(*autoExpander).convert()
      <autogenerated>:1 +0x88
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.autoFlexConvert()
      /Users/jaredbaker/development/terraform-provider-aws/internal/framework/flex/autoflex.go:54 +0x2a4
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.Expand()
      /Users/jaredbaker/development/terraform-provider-aws/internal/framework/flex/auto_expand.go:35 +0xc8
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.runAutoExpandTestCases.func1()
      /Users/jaredbaker/development/terraform-provider-aws/internal/framework/flex/auto_expand_test.go:895 +0xcc
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1689 +0x180
  testing.(*T).Run.gowrap1()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1742 +0x40

Previous read at 0x00c00051c780 by goroutine 20:
  reflect.Value.IsNil()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/reflect/value.go:1572 +0xd4
  github.com/google/go-cmp/cmp.(*state).compareSlice()
      /Users/jaredbaker/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:420 +0x30
  github.com/google/go-cmp/cmp.(*state).compareAny()
      /Users/jaredbaker/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:291 +0xf70
  github.com/google/go-cmp/cmp.(*state).comparePtr()
      /Users/jaredbaker/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:566 +0x698
  github.com/google/go-cmp/cmp.(*state).compareAny()
      /Users/jaredbaker/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:295 +0xf44
  github.com/google/go-cmp/cmp.Diff()
      /Users/jaredbaker/go/pkg/mod/github.com/google/go-cmp@v0.6.0/cmp/compare.go:122 +0x94
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.runAutoExpandTestCases.func1()
      /Users/jaredbaker/development/terraform-provider-aws/internal/framework/flex/auto_expand_test.go:906 +0x268
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1689 +0x180
  testing.(*T).Run.gowrap1()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1742 +0x40

Goroutine 19 (running) created at:
  testing.(*T).Run()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1742 +0x5e4
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.runAutoExpandTestCases()
      /Users/jaredbaker/development/terraform-provider-aws/internal/framework/flex/auto_expand_test.go:887 +0x88
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.TestExpandListOfStringEnum()
      /Users/jaredbaker/development/terraform-provider-aws/internal/framework/flex/auto_expand_test.go:768 +0x488
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1689 +0x180
  testing.(*T).Run.gowrap1()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1742 +0x40

Goroutine 20 (finished) created at:
  testing.(*T).Run()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1742 +0x5e4
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.runAutoExpandTestCases()
      /Users/jaredbaker/development/terraform-provider-aws/internal/framework/flex/auto_expand_test.go:887 +0x88
  github.com/hashicorp/terraform-provider-aws/internal/framework/flex.TestExpandListOfStringEnum()
      /Users/jaredbaker/development/terraform-provider-aws/internal/framework/flex/auto_expand_test.go:768 +0x488
  testing.tRunner()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1689 +0x180
  testing.(*T).Run.gowrap1()
      /opt/homebrew/Cellar/go/1.22.2/libexec/src/testing/testing.go:1742 +0x40
==================
    testing.go:1398: race detected during execution of test
--- FAIL: TestExpandListOfStringEnum (0.00s)
    --- PASS: TestExpandListOfStringEnum/null_value (0.00s)
    --- PASS: TestExpandListOfStringEnum/valid_value (0.00s)
    --- FAIL: TestExpandListOfStringEnum/empty_value (0.00s)
FAIL
FAIL    github.com/hashicorp/terraform-provider-aws/internal/framework/flex     0.496s
FAIL
```

--- a/internal/framework/flex/auto_expand_test.go
+++ b/internal/framework/flex/auto_expand_test.go
@@ -738,9 +738,6 @@ func TestExpandListOfStringEnum(t *testing.T) {
 	var testEnumFoo testEnum = "foo"
 	var testEnumBar testEnum = "bar"
 
-	var testEnums []testEnum
-	testEnumsWant := []testEnum{testEnumFoo, testEnumBar}
-
 	ctx := context.Background()
 	testCases := autoFlexTestCases{
 		{
@@ -749,20 +746,20 @@ func TestExpandListOfStringEnum(t *testing.T) {
 				types.StringValue(string(testEnumFoo)),
 				types.StringValue(string(testEnumBar)),
 			}),
-			Target:     &testEnums,
-			WantTarget: &testEnumsWant,
+			Target:     &[]testEnum{},
+			WantTarget: &[]testEnum{testEnumFoo, testEnumBar},
 		},
 		{
 			TestName:   "empty value",
 			Source:     types.ListValueMust(types.StringType, []attr.Value{}),
-			Target:     &testEnums,
-			WantTarget: &testEnums,
+			Target:     &[]testEnum{},
+			WantTarget: &[]testEnum{},
 		},
 		{
 			TestName:   "null value",
 			Source:     types.ListNull(types.StringType),
-			Target:     &testEnums,
-			WantTarget: &testEnums,
+			Target:     &[]testEnum{},
+			WantTarget: &[]testEnum{},
 		},
 	}
 	runAutoExpandTestCases(ctx, t, testCases)
@@ -775,9 +772,6 @@ func TestExpandSetOfStringEnum(t *testing.T) {
 	var testEnumFoo testEnum = "foo"
 	var testEnumBar testEnum = "bar"
 
-	var testEnums []testEnum
-	testEnumsWant := []testEnum{testEnumFoo, testEnumBar}
-
 	ctx := context.Background()
 	testCases := autoFlexTestCases{
 		{
@@ -786,20 +780,20 @@ func TestExpandSetOfStringEnum(t *testing.T) {
 				types.StringValue(string(testEnumFoo)),
 				types.StringValue(string(testEnumBar)),
 			}),
-			Target:     &testEnums,
-			WantTarget: &testEnumsWant,
+			Target:     &[]testEnum{},
+			WantTarget: &[]testEnum{testEnumFoo, testEnumBar},
 		},
 		{
 			TestName:   "empty value",
 			Source:     types.SetValueMust(types.StringType, []attr.Value{}),
-			Target:     &testEnums,
-			WantTarget: &testEnums,
+			Target:     &[]testEnum{},
+			WantTarget: &[]testEnum{},
 		},
 		{
 			TestName:   "null value",
 			Source:     types.SetNull(types.StringType),
-			Target:     &testEnums,
-			WantTarget: &testEnums,
+			Target:     &[]testEnum{},
+			WantTarget: &[]testEnum{},
 		},
 	}
 	runAutoExpandTestCases(ctx, t, testCases)

